### ISSUE TITLE
SC2046 nested printf "%s" "$( $( ) )"

### ISSUE THREAD

**Issue by amydotsh** [koalaman/shellcheck#2451]
#### For bugs
- SC2046
- 0.8.0
- [y] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [y] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh
printf "%s\n" "shell: $(ps -p $(ps -p $$ o ppid=) o args=)"

```

#### Here's what shellcheck currently says:

```
In test line 2:
printf "%s\n" "shell: $(ps -p $(ps -p $$ o ppid=) o args=)"
                              ^-----------------^ SC2046 (warning): Quote this to prevent word splitting.
```

#### Here's what I wanted or expected to see:
not that? it breaks the script, unless im missing something.

**Comment by leagris** [comment]
Your code is invalid ans shellcheck is right.

Fixed your code as:
```sh
printf "shell: %s\n" "$(ps -p $(("$(ps -p $$ o ppid=)")) o args=)"
```

**Comment by koalaman** [comment]
In general, you definitely have to quote $(..) inside $(..), so the warning is not wrong. In this specific case, Linux (procps-ng) `ps` outputs the value padded with spaces, which therefore can't be reused as input, making it more awkward than normal.

You were relying on word splitting to remove these spaces, which is realistically ok in this specific case, though you can also consider `$((..))` as suggested here, or simply use `$PPID`. 

There's also `ppid:1=` to avoid padding the field so that plain quoting works, but this is Linux-specific and will e.g. fail on macOS where `ps` does not output the padding in the first place anyways.
